### PR TITLE
Add image capture and upload support

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

**Comment by jbuck** [comment]
@k88hudson r?

File: views/share/index.js [hunk 1]
```diff
@@ -1,11 +1,94 @@
...
+                var request_body = {};
+                if (config.PUBLISH_DEV_MODE) request_body.user = user;
+
+                xhr({
+                    method: 'POST',
+                    url: config.IMAGE_REQUEST_ENDPOINT,
+                    json: request_body,
+                    withCredentials: true
+                }, function (wp_error, response, data) {
+                    data.starts_with.forEach(function (field) {
```
Comment (k88hudson) [inline views/share/index.js]: this is dangerous, if data.starts_with doesn't exist. which it doesnt if there's an error

File: views/share/index.js [hunk 2]
```diff
@@ -1,11 +1,94 @@
...
+                        form.append(field, data.fields[field]);
+                    });
+                    form.append("file", file);
+
+                    xhr({
+                        method: 'POST',
+                        url: data.host,
+                        body: form
+                    }, function (s3_error, response, data) {
+                        block.attributes.image.url =
```
Comment (k88hudson) [inline views/share/index.js]: handle error please

File: lib/local-cache.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    getFile: function (key, callback) {
+        // Do we already have a blob URL that we can use?
+        if (URLCache[key]) {
+            return process.nextTick(function () {
+                callback(null, URLCache[key]);
+            });
+        }
+
+        localforage.getItem(key, function (err, file) {
+            if (!file) {
```
Comment (k88hudson) [inline lib/local-cache.js]: handle error as well please

**Comment by k88hudson** [comment]
can you also update the templates to have the new schema? thanks

File: views/share/index.js [hunk 4]
```diff
@@ -1,11 +1,94 @@
...
+var xhr = require('xhr');
 var app;
 
 var PUBLISH_TIMEOUT = 20000;
 
+var pre_publish_hooks = [
+    function image_upload(app, user, hook_callback) {
+        var app_data = app.data;
+        var images = app_data.blocks.filter(function (block) {
+            return block.type === 'image' && block.attributes.image.hash;
```
Comment (k88hudson) [inline views/share/index.js]: might be a good idea to check `block.attributes.image` as well

**Comment by k88hudson** [comment]
Great start. needs some careful handling of error states

**Pull request closed without merging**